src/utils/model-utils.ts

```diff
@@ -0,0 +1,27 @@
+import type { ModelStatic } from '../model';
+import { Model } from '../model';
+
+/**
+ * Returns true if the value is a model subclass.
+ *
+ * @param val The value whose type will be checked
+ */
+export function isModelStatic<M extends Model>(val: any): val is ModelStatic<M> {
+  return typeof val === 'function' && val.prototype instanceof Model;
+}
+
+/**
+ * Returns true if a & b are the same model.
```

Comment:
Do you want to add some extra info here since we are checking for the same initial model?

Reply:
See if the updated description is better